### PR TITLE
NDEF scanning

### PR DESCRIPTION
**Pull request by jesserockz**

### CODE REVIEW

File: includes/ndef.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ public:
+  NDEFTagReader() : PollingComponent(1000) {}
+  void setup() override {
+    this->pn532spi_ = new PN532_SPI(SPI, 0);
+    this->nfc_ = new NfcAdapter(*this->pn532spi_);
+    this->nfc_->begin();
+  }
+
+  void update() override {
+    if (this->nfc_->tagPresent()) {
```
Comment (balloob) [inline includes/ndef.h:23]: Let's rewrite to use a guard clause. So if not present: return. That way code is easier to read.

File: tagreader.yaml [hunk 2]
```diff
@@ -68,40 +70,14 @@ ota:
...
-          switch.is_on: led_enabled
-        then:
-        - light.turn_on:
-            id: activity_led
-            brightness: 100%
-            red: 0%
-            green: 100%
-            blue: 0%
-            flash_length: 500ms
+  clk_pin: D5
```
Comment (balloob) [inline tagreader.yaml:73]: Can we set these via substitution so it's easier to customize

File: tagreader.yaml [hunk 3]
```diff
@@ -17,8 +17,11 @@ esphome:
   name: $devicename
   platform: ESP8266
   board: d1_mini
-  includes: 
-    - rfidreader/src/test.h
+  includes:
+  - includes/ndef.h
+  libraries:
+  - https://github.com/jesserockz/NDEF.git
```
Comment (balloob) [inline tagreader.yaml:23]: We should try to upstream the fixes so we in the future no longer need to rely on forks

Comment (jesserockz) [reply tagreader.yaml:23]: The "fixes" are stripping out the files that cause ESPHome compile warnings and errors... 
Probably not something that can be upstreamed... 
Given time I may be able to integrate it directly into esphome. 
This is more of a quick win to get it out before they are all sent out pre-flashed.

File: tagreader.yaml (hunk 2 above)
Comment (jesserockz) [reply tagreader.yaml:73]: We could yes, maybe as part of another PR then with a bit of tidy up

File: includes/ndef.h (hunk 1 above)
Comment (jesserockz) [reply includes/ndef.h:23]: Can do this

File: tagreader.yaml (hunk 2 above)
Comment (adonno) [reply tagreader.yaml:73]: why use substitutions I understand in the case of names that you would have to type several times 
But where is the benefit here?

Comment (balloob) [reply tagreader.yaml:73]: You can keep using the original YAML but still have a different configuration for your board. It's about splitting configuration (which pin is connected to what) and what to do with each pin (the YAML)

**Pull request merged**